public/app/features/dashboard-scene/scene/LibraryVizPanel.tsx

```diff
@@ -30,6 +32,10 @@ export class LibraryVizPanel extends SceneObjectBase<LibraryVizPanelState> {
         menu: new VizPanelMenu({
           $behaviors: [panelMenuBehavior],
         }),
+        titleItems: [
+          new VizPanelLinks({ rawLinks: [], menu: new VizPanelLinksMenu({ $behaviors: [panelLinksBehavior] }) }),
+          new PanelNotices(),
+        ],
```

Comment:
this constructor looks wrong to me (Where these lines are). The constructor ignores a bunch of state props, so won't work when cloned, this needs a rethink. 

Need test this a bit

Reply:
Yes this is a bit weird, and something that i don't fully understand.
I added this because if the VizPanel doesn't have the VizPanelLinks it errors out here: https://github.com/grafana/grafana/blob/e0311c9771b6e0c89baa4287ab1eb55d49a90161/public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.ts#L242

My thought was to add empty links to this vizPanel that are then replaced by the real links when the library panel loads.

Reply:
@oscarkilhed when is that called before the library panel is loaded?